sdk/src/client/api/block_builder/input_selection/core/remainder.rs

```diff
@@ -22,8 +22,26 @@ use crate::{
 impl InputSelection {
     // Gets the remainder address from configuration of finds one from the inputs.
     fn get_remainder_address(&self) -> Option<(Address, Option<Bip44>)> {
-        if self.remainder_address.is_some() {
-            return self.remainder_address.map(|address| (address, None));
+        if let Some(remainder_address) = self.remainder_address {
+            // Search in inputs for the Bip44 chain for the remainder address, so the ledger can regenerate it
+            #[cfg(feature = "ledger_nano")]
```

Comment:
I wonder if we should just do it always

Reply:
Could be useful for other contexts than ledger_nano idk

Reply:
True